src/app/branded/branded-checkout.component.js

```diff
@@ -78,17 +79,43 @@ class BrandedCheckoutController {
         this.checkoutStep = 'thankYou'
         break
     }
-    this.$window.document.querySelector('branded-checkout').scrollIntoView({ behavior: 'smooth' })
+    this.$element.querySelector('branded-checkout').scrollIntoView({ behavior: 'smooth' })
   }
 
-  previous () {
-    switch (this.checkoutStep) {
-      case 'review':
-        this.fireAnalyticsEvents('contact', 'payment')
-        this.checkoutStep = 'giftContactPayment'
-        break
+  previous (newStep) {
+    let scrollElement = 'branded-checkout'
+
+    if (this.checkoutStep === 'review') {
+      this.fireAnalyticsEvents('contact', 'payment')
+      this.checkoutStep = 'giftContactPayment'
+
+      switch (newStep) {
+        case 'contact':
+          scrollElement = 'contact-info'
+          break
+        case 'cart':
+          scrollElement = 'product-config-form'
+          break
+        case 'payment':
+          scrollElement = 'checkout-existing-payment-methods'
+          break
+      }
     }
-    this.$window.document.querySelector('branded-checkout').scrollIntoView({ behavior: 'smooth' })
+
+    // Watch for changes until the element we are scrolling to exists and everything has loaded
+    // because there will be layout shift every time a new component finishes loading
```

Comment:
Hmm, my experience has always been that the loading indicator remains until all of the pieces have finished loading. Did you run into other cases in your testing?